client/js/socket-events/configuration.js

```diff
@@ -25,6 +25,12 @@ socket.on("configuration", function(data) {
 	options.initialize();
 	webpush.initialize();
 
+	// If localStorage contains a theme that does not exist on this server, switch
+	// back to its default theme.
+	if (!data.themes.map((t) => t.name).includes(options.settings.theme)) {
+		$("#theme-select").val(data.defaultTheme).trigger("change", {force: true});
```

Comment:
This does not belong here but in options.js I think

Reply:
That's awesome, thanks for the feedback. Definitely better than my stupid `force` stuff if I can make it work, I simply missed that capability 😅

Reply:
That worked well, thanks!

@creesch, totally outside of this PR, but just a thought: what would you think about changing `updateSetting`'s signature from `updateSetting(name, value, sync)` to updateSetting(name, value, {sync})`?
That would allow:
```js
options.processSetting("theme", data.defaultTheme, {sync: true});
```

It's a silly change so no big deal, it's just that boolean arguments can be confusing and sometimes error-prone. Definitely more of a style thing, I was just curious about what your opinion on this was.

Reply:
Uh yeah that would be fine I guess. 
Though I am not sure how that would apply to `processSetting()` as the boolean there is `save` ;) 

Looking at it `processSetting()` should probably be restructured a little bit with an options parameter instead so it can be called like this. 

```javascript
processSetting(name, value, {
    save: false,
    sync: false
})
```

Making the declaration something like this 

```javascript
processSetting(name, value, {save, sync}) { 
    // code
}
```

Because the way it is currently written it will not sync the theme value.  

I'll work that all in after this gets merged. Also I'll leave a bit more comments to make it clear to people how options function.

Reply:
Sounds good, thanks for your efforts @creesch!